File: LeetCode/49. Group Anagrams.cpp
// Author : @AdarshGaur

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> mp;
        for(string &s: strs){
            string k = s;
            sort(k.begin(), k.end());
            mp[k].push_back(s);
        }
        vector<vector<string>> ans;
        for(auto &[s, v]: mp){
            ans.push_back(v);
        }
        return ans;
    }
};

File: LeetCode/49. Group Anagrams.py
#Author: @AdarshGaur

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = {}
        ans = []
        last = 0
        for i, s in enumerate(strs):
            new_s = ''.join(sorted(s))
            # print(type(new_s))
            if d.get(new_s) != None:
                ans[d[new_s]].extend([s])
            else:
                ans.append([s])
                d[new_s] = last
                last += 1
        return ans
